src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/dates/ReleaseDateService.kt

```diff
@@ -106,15 +106,20 @@ class ReleaseDateService(
   fun getLicenceStartDate(
     nomisRecord: PrisonerSearchPrisoner,
     licenceKind: LicenceKind? = null,
-  ): LocalDate? = when (
-    licenceKind ?: determineReleaseDateKind(
+  ): LocalDate? {
+    val lsdKind = licenceKind ?: determineReleaseDateKind(
       nomisRecord.postRecallReleaseDate,
       nomisRecord.conditionalReleaseDate,
     )
-  ) {
-    HDC -> nomisRecord.homeDetentionCurfewActualDate
-    PRRD -> nomisRecord.calculatePrrdLicenceStartDate()
-    else -> calculateCrdLicenceStartDate(nomisRecord, iS91DeterminationService.isIS91Case(nomisRecord))
+
+    return when (lsdKind) {
+      HDC -> nomisRecord.homeDetentionCurfewActualDate
+      PRRD -> nomisRecord.calculatePrrdLicenceStartDate()
+      else -> calculateCrdLicenceStartDate(
+        nomisRecord,
+        iS91DeterminationService.isIS91Case(nomisRecord),
+      )
+    }
```

Comment:
I personally disagree that this refactor is necessary, maybe one to chat about as a team to find the style that we prefer? If majority prefer having the assignment separately, I'm happy to move that way in future, but my view is that it's neater without